File: pset6/readability/readability.py
import cs50 as cs
#import ctype as ctype
#import stdio as stdio
#import string as string
#import math as math

sentence = cs.get_string("Text: ")

sentence = sentence.upper()

# sentences
S = 0
# letters
L = 0
# words
W = 1

# loops through every letter in sentence
for i in range(len(sentence)):

    # if it is uppercase (if its a letter), add 1 to letter count
    if(sentence[i].isupper()):
        L += 1
    # if there is a space, add 1 to word count
    if(sentence[i] == ' '):
        W += 1
    # if there is punctuation, add 1 to sentence count
    if(sentence[i] == '.' or sentence[i] == '!' or sentence[i] == '?'):
        S += 1

# set coefficient
coefficient = 100 / W

# set letters per 100 using coefficient
L *= coefficient
S *= coefficient

# set grade as rounded int of grade formula
grade = int(round(0.0588 * L - 0.296 * S - 15.8))

if grade < 1:
    print("Before Grade 1")
    exit()
if grade > 16:
    print("Grade 16+")
    exit()
else:
    grade = str(grade)
    print("Grade " + grade)

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //get sentence
    string sentence = get_string("Text: ");

    float grade = 0;
    float S = 0;
    float L = 0;
    float W = 1;

    for (int i = 0; i < strlen(sentence); i++)
    {
        //changes letter to uppercase
        sentence[i] = toupper(sentence[i]);

        // Checks for uppercase (if its a letter it will be)
        if (isupper(sentence[i]))
        {
            // if there is a letter add +1 to letter list
            L++;
        }
        if (sentence[i] == ' ')
        {
            // if there is a space add +1 to wordslist
            W++;
        }
        if ((sentence[i] == '.') || (sentence[i] == '!') || (sentence[i] == '?'))
        {
            // if it isnt a space or a letter it is punctuation
            S++;
        }
    }
    // prints word, sentence, letter count
//        printf("L is %f, S is %f, W is %f\n", L, S, W);

    //sets coefficient out of 100
    float coefficient = 100 / W;

    //set L and S per 100 words
    L = L * coefficient;
    S = S * coefficient;

    // prints values out of 100
//        printf("L is %f, S is %f, W is %f\n", L, S, W);

    // finds grade
    grade = 0.0588 * L - 0.296 * S - 15.8;

    //rounds grade
    float grade1 = roundf(grade);

    // converts grade to int
    int gradeF = grade1;

    // prints output
    if (gradeF < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    if (gradeF > 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", gradeF);
        return 0;
    }
}